common/src/api/external/mod.rs

```diff
@@ -3090,6 +3090,24 @@ pub enum ImportExportPolicy {
     Allow(Vec<oxnet::IpNet>),
 }
 
+/// Query parameters for reading the support bundle
+#[derive(Deserialize, Serialize, JsonSchema)]
+pub struct SupportBundleGetQueryParams {
+    pub query_type: SupportBundleQueryType,
+}
+
+/// Describes the type of access to the support bundle
+#[derive(Clone, Debug, Deserialize, Serialize, JsonSchema)]
+#[serde(tag = "type", rename_all = "snake_case")]
+pub enum SupportBundleQueryType {
+    /// Access the whole support bundle
+    Whole,
+    /// Access the names of all files within the support bundle
+    Index,
```

Comment:
how does this differ from the HEAD response?

Reply:
Good question - the HEAD response is currently only returning headers indicating the length of the file. It is not returning any extra information about "what files exist in this zipfile".

In the implementation, HEAD is implemented in Nexus by calling the Sled Agent which actually holds the bundle, it calls into `make_head_response` here:

https://github.com/oxidecomputer/omicron/blob/main/range-requests/src/lib.rs#L98-L129